clang/lib/Frontend/CompileJobCache.cpp

```diff
@@ -293,11 +300,27 @@ std::optional<int> CompileJobCache::initialize(CompilerInstance &Clang) {
   if (!CAS || !Cache)
     return 1; // Exit with error!
 
+  CompilerInvocation DummyInvocation = Invocation;
+  CompileJobCachingOptions DummyCacheOpts;
+
+  std::string CASInputFileID;
+  std::swap(CASInputFileID, Invocation.getFrontendOpts().CASInputFileID);
   CompileJobCachingOptions CacheOpts;
   ResultCacheKey =
       canonicalizeAndCreateCacheKey(*CAS, Diags, Invocation, CacheOpts);
```

Comment:
I wonder if we should sink the multiple key handling into `canonicalizeAndCreateCacheKey`. Was there a reason you didn't do that?

Reply:
I didn't think of that, I'll try it out and see how the API feels.